### PR TITLE
[SPARK-7738] [SQL] [PySpark] add reader and writer API in Python

### PR DESCRIPTION
**Pull request by davies**
cc @rxin, please take a quick look, I'm working on tests.

### CODE REVIEW

**Comment by andrewor14** [comment]
retest this please

File: core/src/main/scala/org/apache/spark/api/python/PythonUtils.scala [hunk 1]
```diff
@@ -50,8 +50,15 @@ private[spark] object PythonUtils {
   /**
    * Convert list of T into seq of T (for calling API with varargs)
    */
-  def toSeq[T](cols: JList[T]): Seq[T] = {
-    cols.toList.toSeq
+  def toSeq[T](vs: JList[T]): Seq[T] = {
```
Comment (rxin) [inline core/src/main/scala/org/apache/spark/api/python/PythonUtils.scala]: what does vs mean?

File: python/pyspark/sql/readwriter.py [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+        if format is not None:
+            jwrite = jwrite.format(format)
+        for k in options:
+            jwrite = jwrite.option(k, options[k])
+        if path is None:
+            jwrite.save()
+        else:
+            jwrite.save(path)
+
+    def saveAsTable(self, name, format=None, mode="error", **options):
```
Comment (rxin) [inline python/pyspark/sql/readwriter.py]: are there any test cases covering this?

**Comment by davies** [comment]
@rxin @yhai How do we test JDBC?

**Comment by jkbradley** [comment]
@davies I think this has the wrong JIRA number in the title.

File: python/pyspark/sql/readwriter.py (hunk 2 above)
Comment (davies) [inline python/pyspark/sql/readwriter.py]: It's covered by df.saveAsTable(), had updated the tests to use new API.

**Comment by davies** [comment]
@jkbradley, thanks, corrected.

File: core/src/main/scala/org/apache/spark/api/python/PythonUtils.scala (hunk 1 above)
Comment (davies) [inline core/src/main/scala/org/apache/spark/api/python/PythonUtils.scala]: value -> v, values => vs :)

**Pull request closed without merging**